## Differential Gene Expression Analysis between Mouse Sham and TAC RNA-seq
  
  
### Input  
Input file: countfile.txt (hard-coded for now)      
This file is a tab-delimited text file with 3 columns. Header is optional.    
| Column   | Description |
| -------- | ----------- |
| Column 1 | full path to the count file generated by htseq-count |
| Column 2 | sample id                                            |
| Column 3 | condition/grouping information used for DE analysis  |  
  
Example countfile.txt:  
| countfile | sampleid | group |
| --- | --- | --- |
| RMH3869_count.txt | RMH3869 | TAC |
| RMH3871_count.txt | RMH3871 | SHAM |
| RMH3872_count.txt | RMH3872 | TAC |
| RMH3874_count.txt | RMH3874 | SHAM |
| RMH3875_count.txt | RMH3875 | TAC |
| RMH3877_count.txt | RMH3877 | SHAM |

      

### Command  
I usualy run my R script on linux command line. You can also modify the script to make it work on Rstudio or any IDE.  
The command line will require two user arguments for this particular script. These arguments are the sample groupings. In my case, it will be SHAM and TAC.  
I am usually R version 3.5.1.  
```
Rscript-3.5.1 single_edgeR.r SHAM TAC
```
  

### Output  
The R script will generate a few output files.  

1. de_SHAM_TAC.txt  
This is the output of differential gene expression test performed by edgeR. Although there is no definite cut-off to define significance, I use abs(log2FC)>0.5, FDR<0.05 and avg FPKM>1 as a cut-off for significant DEGs. The columns are:  


| Column | Description            |
| :------: | -----------            |
| 1      | GENCODE Gene ID        |
| 2      | log2FC (Group2/Group1) |
| 3 | logCPM across the samples |
| 4 | Test statistics (loglikelihodd) |
| 5 | P-value |
| 6 | FDR |
| 7 | Gene Symbol / HGNC |
| 8 | Biotype |
| 9-whatever | FPKM of all samples |


2. Scatterplot (scatterplot_withtext_SHAM_TAC.png)  
One of the plots that we can look at is scatterplot that represents the correlation between the two conditions (SHAM and TAC). The x-axis in the graph below shows average FPKM of SHAM cardiomyocytes and the y-axis shows average FPKM of TAC cardiomyocytes.  

![alt text](https://github.com/lwtan90/wilsonNGS/blob/master/RNAseq/testdata/SHAMTAC/scatterplot_withtext_SHAM_TAC.png )


3. Volcano Plot (volcanoplot_withtext_SHAM_TAC.png)  

![alt text](https://github.com/lwtan90/wilsonNGS/blob/master/RNAseq/testdata/SHAMTAC/volcanoplot_withtext_SHAM_TAC.png )

4. GSEA analysis (using FGSEA R package)  
Two files per pathway DB tested (GO_analysis.txt or GO.RData)  

  
Upregulated Pathways:  
  
| pathway | pval | padj | ES | NES |
| :--- | :--: | :--: | :--: | :--: |
| GO_COLLAGEN_TRIMER | 0 | 0.0079 | 0.8466 | 2.682 |
| GO_EXTRACELLULAR_MATRIX_COMPONENT | 0 | 0.0079 | 0.7538 | 2.6497 |
| GO_PROTEINACEOUS_EXTRACELLULAR_MATRIX | 0 | 0.0079 | 0.7161 | 2.6478 |
| GO_EXTRACELLULAR_STRUCTURE_ORGANIZATION | 0 | 0.0079 | 0.7128 | 2.6383 |
| GO_EXTRACELLULAR_MATRIX_BINDING | 0 | 0.0079 | 0.8044 | 2.6162 |
| GO_EXTRACELLULAR_MATRIX_STRUCTURAL_CONSTITUENT | 0 | 0.0079 | 0.8255 | 2.6064 |
| GO_EXTRACELLULAR_MATRIX | 0 | 0.0079 | 0.6939 | 2.6011 |
| GO_BASEMENT_MEMBRANE | 0 | 0.0079 | 0.7501 | 2.5799 |
| GO_COLLAGEN_BINDING | 0 | 0.0079 | 0.7807 | 2.564 |
  
  
Downregulated Pathways:  
  

| pathway | pval | padj | ES | NES |
| :--- | :--: | :--: | :--: | :--: |
| GO_MITOCHONDRIAL_ATP_SYNTHESIS_COUPLED_PROTON_TRANSPORT | 0 | 0.0143 | -0.7478 | -2.3421 |
| GO_PROTON_TRANSPORTING_ATP_SYNTHASE_COMPLEX | 0 | 0.0148 | -0.6904 | -2.2669 |
| GO_BRANCHED_CHAIN_AMINO_ACID_METABOLIC_PROCESS | 0 | 0.0148 | -0.6333 | -2.0795 |
| GO_PHOTORECEPTOR_CELL_MAINTENANCE | 0 | 0.016 | -0.5669 | -1.9882 |
| GO_MALE_MEIOSIS | 0 | 0.016 | -0.5459 | -1.9145 |
| GO_NADH_DEHYDROGENASE_ACTIVITY | 0 | 0.0172 | -0.6156 | -2.2153 |
| GO_MITOCHONDRIAL_ELECTRON_TRANSPORT_NADH_TO_UBIQUINONE | 0 | 0.0179 | -0.6089 | -2.2264 |
| GO_NADH_DEHYDROGENASE_COMPLEX | 0 | 0.0179 | -0.5823 | -2.1291 |
| GO_FATTY_ACID_BETA_OXIDATION | 0 | 0.0183 | -0.4836 | -1.8241 |
