## Differential Gene Expression Analysis between Mouse Sham and TAC RNA-seq
  
  
### Input  
Input file: countfile.txt (hard-coded for now)      
This file is a tab-delimited text file with 3 columns. Header is optional.    
| Column   | Description |
| -------- | ----------- |
| Column 1 | full path to the count file generated by htseq-count |
| Column 2 | sample id                                            |
| Column 3 | condition/grouping information used for DE analysis  |  
  
Example countfile.txt:  
| countfile | sampleid | group |
| --- | --- | --- |
| RMH3869_count.txt | RMH3869 | TAC |
| RMH3871_count.txt | RMH3871 | SHAM |
| RMH3872_count.txt | RMH3872 | TAC |
| RMH3874_count.txt | RMH3874 | SHAM |
| RMH3875_count.txt | RMH3875 | TAC |
| RMH3877_count.txt | RMH3877 | SHAM |

      

### Command  
I usualy run my R script on linux command line. You can also modify the script to make it work on Rstudio or any IDE.  
The command line will require two user arguments for this particular script. These arguments are the sample groupings. In my case, it will be SHAM and TAC.  
I am usually R version 3.5.1.  
```
Rscript-3.5.1 single_edgeR.r SHAM TAC
```
  

### Output  
The R script will generate a few output files.  

1. de_SHAM_TAC.txt
This is the output of differential gene expression test performed by edgeR. The columns are:  


| Column | Description            |
| :------: | -----------            |
| 1      | GENCODE Gene ID        |
| 2      | log2FC (Group2/Group1) |
| 3 | logCPM across the samples |
| 4 | Test statistics (loglikelihodd) |
| 5 | P-value |
| 6 | FDR |
| 7 | Gene Symbol / HGNC |
| 8 | Biotype |
| 9-whatever | FPKM of all samples |

